logstash-core/lib/logstash/settings.rb

```diff
@@ -557,6 +560,54 @@ def coerce(value)
         end
       end
     end
+
+    class CloudId < Coercible
+      def initialize(name, default, strict=true, &validator_proc)
+        super(name, Util::CloudId, default, false, &validator_proc)
+      end
+
+      def set(value)
+        @value = coerce(value)
+        @value_is_set = true
+        @value
+      end
+
+      def coerce(value)
+        if value.is_a?(@klass)
+          return value
+        end
+        @klass.new(value)
+      end
+
+      protected
+      def validate(value)
+        coerce(value)
+      end
+    end
+
+    class CloudAuth < Coercible
+      def initialize(name, default, strict=true, &validator_proc)
+        super(name, Util::CloudAuth, default, false, &validator_proc)
+      end
+
+      def set(value)
+        @value = coerce(value)
+        @value_is_set = true
+        @value
+      end
+
+      def coerce(value)
+        if value.is_a?(@klass)
+          return value
+        end
+        @klass.new(value)
+      end
+
+      protected
+      def validate(value)
+        coerce(value)
+      end
+    end
   end
```

Comment:
I can!!
I'll call the merged class CloudCredential and pass in the appropriate class